src/main/java/reposense/authorship/FileInfoExtractor.java

```diff
@@ -35,19 +52,101 @@
         // checks out to the latest commit of the date range to ensure the FileInfo generated correspond to the
         // git blame file analyze output
         GitChecker.checkoutToDate(config.getRepoRoot(), config.getBranch(), config.getUntilDate());
-        ArrayList<FileInfo> fileInfos = new ArrayList<>();
-        getAllFileInfo(config, Paths.get(config.getRepoRoot()), fileInfos);
+        List<FileInfo> fileInfos = new ArrayList<>();
+        String lastCommitHash = CommandRunner.getCommitHashBeforeDate(
+                config.getRepoRoot(), config.getBranch(), config.getSinceDate());
+
+        if (!lastCommitHash.isEmpty()) {
+            fileInfos = getEditedFileInfos(config, lastCommitHash);
+        } else {
+            getAllFileInfo(config, Paths.get(config.getRepoRoot()), fileInfos);
+        }
+
         fileInfos.sort(Comparator.comparing(FileInfo::getPath));
+        return fileInfos;
+    }
+
+    /**
+     * Generates a list of relevant {@code FileInfo} for all files that were edited in between the current
+     * commit and the {@code lastCommitHash} commit, marks each {@code LineInfo} for each {@code FileInfo} on
+     * whether they have been inserted within the commit range or not, and returns it.
+     */
+    public static List<FileInfo> getEditedFileInfos(RepoConfiguration config, String lastCommitHash) {
+        List<FileInfo> fileInfos = new ArrayList<>();
+
+        String diffResult = CommandRunner.diffCommit(config.getRepoRoot(), lastCommitHash);
+
+        // no diff between the 2 commits, return an empty list
+        if (diffResult.isEmpty()) {
+            return fileInfos;
+        }
+
+        String[] fileDiffResultList = diffResult.split(DIFF_FILE_CHUNK_SEPARATOR);
+
+        for (String fileDiffResult : fileDiffResultList) {
+            String[] fileDiffResultLines = fileDiffResult.split("\n");
+            String[] filesCompared = fileDiffResultLines[COMPARED_FILE_LINE_INDEX].split(COMPARED_FILE_PATH_SEPARATOR);
```

Comment:
don't really get this part. how do you get `files` from `file`?

Reply:
Hmm, the `files` here is for the 'before' and 'after' for a single `file` change, since the file from 'before' may have been renamed into the 'after'.
Maybe I can just mix these with the below line, since we are only interested in the 'after' file name?

Reply:
Also what you want is the first line here... what if the original directory contains `b/`? E.g. `blob/`

```
diff --git a/docs/UserGuide.md b/docs/UserGuide.md
index 6f0b634..a8af6df 100644
--- a/docs/UserGuide.md
+++ b/docs/UserGuide.md
@@ -18 +18 @@
-Usage: `java -jar RepoSense.jar -config CSV_CONFIG_FILE_PATH [-output OUTPUT_DIRECTORY] [-since DD/MM/YYYY] [-until DD/MM/YYYY]`
+Usage: `java -jar RepoSense.jar -config CSV_CONFIG_FILE_PATH [-output OUTPUT_DIRECTORY] [-since DD/MM/YYYY] [-until DD/MM/YYYY] [-formats FORMAT...]`
@@ -25 +25 @@ Sample usage:
-$ java -jar RepoSense.jar -config CSV_path.csv -output output_path/ -since 01/10/2017 -until 01/11/2017
+$ java -jar RepoSense.jar -config CSV_path.csv -output output_path/ -since 01/10/2017 -until 01/11/2017 -formats java adoc js
@@ -31,2 +31,3 @@ Argument List:
-- since : Optional. start date of analysis. Format: `DD/MM/YYYY`
-- until : Optional. end date of analysis. Format: `DD/MM/YYYY`
+- since : Optional. The start date of analysis. Format: `DD/MM/YYYY`
+- until : Optional. The end date of analysis. Format: `DD/MM/YYYY`
+- formats : Optional. The file formats to analyse. Formats: `alphanumerical file formats`. If not provided, the following file formats will be used. `adoc, cs, css, fxml, gradle, html, java, js, json, jsp, md, py, tag, xml`

```

Reply:
Perhaps use regex matcher and capturing group instead to make it cleaner? Also, it seems that there are some left over texts after `DIFF_FILE_CHUNK_SEPARATOR` which cause the following method to skip the first index.